Gems/Atom/Tools/AtomToolsFramework/Code/Source/EntityPreviewViewport/EntityPreviewViewportScene.cpp

```diff
@@ -101,13 +91,96 @@ namespace AtomToolsFramework
         }
     }
 
+    EntityPreviewViewportScene::RenderPipelineMap::iterator EntityPreviewViewportScene::AddRenderPipeline(const AZ::Data::AssetId& pipelineAssetId)
+    {
+        // Load the render pipeline asset
+        AZStd::optional<AZ::RPI::RenderPipelineDescriptor> pipelineDesc =
+            AZ::RPI::GetRenderPipelineDescriptorFromAsset(pipelineAssetId, m_viewportIdSuffix);
+        AZ_Assert(pipelineDesc.has_value(), "Invalid render pipeline descriptor from asset %s", pipelineAssetId.ToFixedString().c_str());
```

Comment:
I wonder if it's worth trying to recover a bit more gracefully in this case, is it possible that pipelines are edited/moved while this tool is running for example?

Reply:
Sure I can improve this a bit.